### PR TITLE
Add prow config for upgrade tests

### PR DESCRIPTION
**Pull request by jonjohnsonjr**
I took a stab at this, but I am not sure how to test that this actually works :)

### CODE REVIEW

**Comment by adrcunha** [comment]
/hold

This won't do what you think it'll do, there are missing pieces.

**Comment by jonjohnsonjr** [comment]
Thanks 😄

File: scripts/presubmit-tests.sh [hunk 1]
```diff
@@ -88,6 +88,10 @@ function main() {
         RUN_INTEGRATION_TESTS=1
         shift
         ;;
+      --upgrade-tests)
```
Comment (adrcunha) [inline scripts/presubmit-tests.sh]: You can revert this file, https://github.com/knative/test-infra/pull/256 implements running arbitrary tests on presubmit.

File: ci/prow/config.yaml [hunk 2]
```diff
@@ -214,6 +214,39 @@ presubmits:
...
+        - "--clean"
+        - "--job=$(JOB_NAME)"
+        - "--repo=github.com/$(REPO_OWNER)/$(REPO_NAME)=$(PULL_REFS)"
+        - "--root=/go/src"
+        - "--service-account=/etc/service-account/service-account.json"
+        - "--upload=gs://knative-prow/pr-logs"
+        - "--" # end bootstrap args, scenario args below
+        - "--" # end kubernetes_execute_bazel flags (consider following flags as text)
+        - "./test/presubmit-tests.sh"
+        - "--upgrade-tests"
```
Comment (adrcunha) [inline ci/prow/config.yaml]:
```suggestion
        - "--run-test"
        - "./test/upgrade-tests.sh"
```

File: ci/prow/config.yaml [hunk 3]
```diff
@@ -214,6 +214,39 @@ presubmits:
           requests:
             memory: "1Gi"
 
+  - name: pull-knative-serving-upgrade-tests
```
Comment (adrcunha) [inline ci/prow/config.yaml]: Also, do we want to run these tests in the CI flow?

Comment (jonjohnsonjr) [reply ci/prow/config.yaml]: I think so -- I'm not sure if we want them to be fatal yet, @mattmoor thoughts?

Comment (jonjohnsonjr) [reply ci/prow/config.yaml]: Matt says yes to CI flow and yes to failures being fatal (i.e. must pass).

**Comment by adrcunha** [comment]
#256 is merged, https://github.com/knative/serving/pull/2494 adds the required infra to the knative/serving repo.

File: ci/prow/config.yaml (hunk 3 above)
Comment (adrcunha) [reply ci/prow/config.yaml]: Ok. https://github.com/knative/serving/pull/2502 will make the upgrade test run on CI and presubmits. Then we'll just have to tweak the Prow jobs and we're done.

File: ci/prow/config.yaml [hunk 4]
```diff
@@ -214,6 +214,40 @@ presubmits:
...
+        - "--job=$(JOB_NAME)"
+        - "--repo=github.com/$(REPO_OWNER)/$(REPO_NAME)=$(PULL_REFS)"
+        - "--root=/go/src"
+        - "--service-account=/etc/service-account/service-account.json"
+        - "--upload=gs://knative-prow/pr-logs"
+        - "--" # end bootstrap args, scenario args below
+        - "--" # end kubernetes_execute_bazel flags (consider following flags as text)
+        - "./test/presubmit-tests.sh"
+        - "--run-test"
+        - "./test/upgrade-tests.sh"
```
Comment (adrcunha) [inline ci/prow/config.yaml]:
```suggestion
        - "./test/e2e-upgrade-tests.sh"
```

File: ci/prow/config.yaml (hunk 3 above)
Comment (adrcunha) [reply ci/prow/config.yaml]: On line 209, please replace `- "--integration-tests"` with:

```
        - "--run-test"
        - "./test/e2e-tests.sh"
```

**Comment by jonjohnsonjr** [comment]
Anything left on this?

**Comment by adrcunha** [comment]
> Anything left on this?

No, the CI flow is fine

https://gubernator.knative.dev/build/knative-prow/logs/ci-knative-serving-continuous/1063175232929402880

so we should be good to go. :)

/hold cancel
/approve
/lgtm

I'll update Prow once this PR merges.

**Pull request merged**